src/Stryker.Core/Stryker.Core/ProjectComponents/ExcludableProjectComponent.cs

```diff
@@ -0,0 +1,119 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using DotNet.Globbing;
+
+namespace Stryker.Core.ProjectComponents
+{
+    public abstract class ExcludableProjectComponent<T, TSpan> : ProjectComponent<T>
```

Comment:
> Abstract classes, the guardians of design,
Blueprints of thought, where concepts align.
Not fully formed, they never instantiate,
Yet from their essence, objects radiate.